Fix #97

Marekkon5 opened the pull request:
Just exposes the `Encoding` enum.

Marekkon5 commented:
I understand your suggestion, however that requires refactoring all the `Decoder` methods to return `Option<Encoding>` as well and saving that into the Frame. Is it what you had in mind?

polyfloyd commented:
Well not all, just the GEOB decoder I would say?

I do not have a lot in mind so if you are able to propose something that supports this usecase without breaking the API I would be happy. Besides, if it does not expose any new fields or functions and the behavior is not too complex we can refactor it later.

Marekkon5 commented:
Thank you, I'll see what I can do tomorrow.

Marekkon5 commented:
I've had to change the `.unique` for frame, because newly created GEOB Frames will have no encoding (and in tests you compare created `None` encoding GEOB to written and read GEOB with default encoding).

Marekkon5 commented:
Hello, I've added one more commit regarding #96 - we've found out that Serato also stores `SERATO_PLAYCOUNT` tag which it can't read unless it's in UTF-8 (ID3v2.4) or Latin1 (ID3v2.3). I've added the encoding preservation for TXXX too. 

Thank you

polyfloyd commented:
Hi, just want to let you know that I have not forgotten about your effort! I expect to have time for this in about 3 weeks

polyfloyd reviewed (commented):
Alright, managed to find a moment to review your changes

Marekkon5 commented on src/frame/mod.rs:
```diff
@@ -193,6 +208,11 @@ impl Frame {
     }
 
+    /// Returns the encoding of this frame
+    pub fn encoding(&self) -> Option<Encoding> {
```
If encoding is set to None, it is unknown until you write the tag (then the default for given ID3v2 version will be used). Same applies for the `.set_encoding()` fn - it is to override the default encoding used by the ID3v2 version.

Marekkon5 commented on src/frame/mod.rs:
```diff
@@ -373,6 +393,16 @@ impl Frame {
     }
 }
 
+impl PartialEq for Frame {
```
It is because of `None` encoding - Frames should match if the encoding is None in one of them. Also without it some tests are broken (if you read Frame from file it WILL have an encoding, but if you create a new frame, it won't have encoding until you write to file (or you manually set it))

polyfloyd replied:
In that case, could we also have this implementation for `Eq`? `Hash` is also a thing that still worries me, although I am not sure how to resolve that. Perhaps leaving it auto-derived is good enough

Marekkon5 replied:
The `Eq` doesn't do anything special, it just informs the compiler that the PartialEq impl is actually Eq. So just deriving it should be enough (source: https://doc.rust-lang.org/std/cmp/trait.Eq.html)

polyfloyd replied:
Ah, good :)

polyfloyd commented on src/frame/mod.rs:
```diff
@@ -113,9 +121,16 @@ impl Frame {
     }
 
+    /// Sets the encoding for this frame.
+    pub fn set_encoding(mut self, encoding: Option<Encoding>) -> Self {
```
Could you still answer this one?

Marekkon5 replied:
I used it in my own project (id3 as a dependency) and I think the user should have the freedom to set the encoding if he needs it for whatever reason.

polyfloyd commented:
Alright, sorry for taking so long. Thank your for picking this up!
